core/common/hexutil.cpp

```diff
@@ -3,27 +3,31 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <algorithm>
-#include <boost/algorithm/hex.hpp>
+#include "common/hexutil.hpp"
 
-#include "hexutil.hpp"
+#include <boost/algorithm/hex.hpp>
 
 namespace kagome::common {
 
   std::string hex(const uint8_t *array, size_t len) noexcept {
     std::string res(len * 2, '\x00');
-    boost::algorithm::hex(array, array + len, res.begin()); // NOLINT
+    boost::algorithm::hex(array, array + len, res.begin());  // NOLINT
     return res;
   }
 
   std::string hex(const std::vector<uint8_t> &bytes) noexcept {
     return hex(bytes.data(), bytes.size());
   }
 
-  std::vector<uint8_t> unhex(const std::string &hex) {
+  expected::Result<std::vector<uint8_t>, std::string> unhex(
+      const std::string &hex) {
     std::vector<uint8_t> blob((hex.size() + 1) / 2);
-    boost::algorithm::unhex(hex.begin(), hex.end(), blob.begin());
-    return blob;
+    try {
+      boost::algorithm::unhex(hex.begin(), hex.end(), blob.begin());
+      return expected::Value{blob};
+    } catch (boost::algorithm::hex_decode_error &e) {
```

Comment:
According to [this](https://stackoverflow.com/questions/2145147/why-catch-an-exception-as-reference-to-const) it's nearly always much better to catch exception by constref, if you are not going to re-throw them to upper levels